src/main/java/com/github/wovnio/wovnjava/Api.java

```diff
@@ -72,15 +74,27 @@ String translate(String lang, String html) throws ApiException, ApiNoPageDataExc
     String translate(String lang, String html, HttpURLConnection con) throws ApiException, ApiNoPageDataException {
         OutputStream out = null;
         try {
-            ByteArrayOutputStream body = gzipStream(getApiBody(lang, html).getBytes());
+            out = con.getOutputStream();
+
+            Map<String, String> apiParams = getApiParameters(lang, html);
+            String urlEncodedBody =  FormUrlEncoding.encode(apiParams);
+            byte[] apiBodyBytes = urlEncodedBody.getBytes();
+
+            if (this.settings.compressApiRequests) {
+                ByteArrayOutputStream compressedBody = gzipStream(apiBodyBytes);
+                compressedBody.writeTo(out);
+                con.setRequestProperty("Content-Type", "application/octet-stream");
+                con.setRequestProperty("Content-Length", String.valueOf(compressedBody.size()));
```

Comment:
Just for general correctness I think we should also include `Content-Encoding: gzip` here.